src/pages/instances/InstanceSearchFilter.tsx

```diff
@@ -44,50 +46,52 @@ const InstanceSearchFilter: FC<Props> = ({ instances }) => {
   ];
 
   const onSearchDataChange = (searchData: SearchAndFilterChip[]) => {
-    const searchTokensByLead = (lead: string) =>
+    const searchValuesByLead = (lead: string) =>
       searchData
         .filter(
           (chip) => chip.lead === lead || (lead === "query" && chip.quoteValue),
         )
         .map((chip) => chip.value);
 
-    const newParams = new URLSearchParams();
-    searchTokensByLead("query").forEach((query) => {
-      newParams.append("query", query);
-    });
-    searchTokensByLead("status").forEach((status) => {
-      newParams.append("status", status);
-    });
-    searchTokensByLead("type").forEach((type) => {
-      newParams.append("type", type);
-    });
-    searchTokensByLead("profile").forEach((profile) => {
-      newParams.append("profile", profile);
+    const newParams = new URLSearchParams(searchParams.toString());
+
+    QUERY_PARAMS.forEach((param) => {
+      const values = searchValuesByLead(param);
+      if (values.length === 0) {
+        newParams.delete(param);
+      }
+      values.forEach((value) => {
+        if (!newParams.has(param, value)) {
+          newParams.append(param, value);
+        }
```

Comment:
When having multiple chips for the same category selected and removing some (but not all), this doesn't remove the value from the params.

I.e. filter "running" or "stopped" and remove "stopped", this last change will not be reflected in the URL params.

Reply:
Good catch! Should be fixed now.